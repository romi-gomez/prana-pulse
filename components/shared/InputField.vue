<template>
  <div>
    <label :for="name">{{ label }}</label>
    <input 
      :id="name" 
      :type="type" 
      :placeholder="placeholder" 
      v-model="inputValue"
      @input="$emit('update:modelValue', inputValue)" 
      class="px-4 py-2 border rounded focus:outline-none focus:ring-2 focus:ring-primary" 
    />
  </div>
</template>

<script setup>
import { defineProps, ref, watch } from 'vue'

const props = defineProps({
  type: {
    type: String,
    default: 'text'
  },
  placeholder: {
    type: String,
    default: ''
  },
  label: {
    type: String,
    required: true
  },
  name: {
    type: String,
    required: true
  },
  modelValue: {
    type: String,
    default: ''
  }
})

const inputValue = ref(props.modelValue)
watch(() => props.modelValue, (newValue) => {
  inputValue.value = newValue
})
</script>

<style scoped>
/* No additional styles needed if using Tailwind */
</style>