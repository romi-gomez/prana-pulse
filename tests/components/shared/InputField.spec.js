import { mount } from '@vue/test-utils'
import InputField from '@/components/shared/InputField.vue'

describe('InputField', () => {
  test('renders correctly', () => {
    const wrapper = mount(InputField, {
      props: {
        label: 'Username',
        name: 'username'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('emits update:modelValue event when input value changes', async () => {
    const wrapper = mount(InputField, {
      props: {
        label: 'Username',
        name: 'username',
        modelValue: ''
      }
    })
    const input = wrapper.find('input')
    await input.setValue('new value')
    expect(wrapper.emitted('update:modelValue')[0]).toEqual(['new value'])
  })
})