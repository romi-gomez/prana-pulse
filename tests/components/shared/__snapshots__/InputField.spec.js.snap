// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders correctly 1`] = `
<div>
  <label
    for="username"
  >
    Username
  </label>
  <input
    class="px-4 py-2 border rounded focus:outline-none focus:ring-2 focus:ring-primary"
    id="username"
    placeholder=""
    type="text"
  />
</div>
`;
